### PR TITLE
Fix forum index showing topics without view other topics permission

### PR DESCRIPTION
**Pull request by samerton**

### CODE REVIEW

File: modules/Forum/classes/Forum.php [hunk 1]
```diff
@@ -808,4 +797,47 @@ public static function getAccessibleLabels(array $labels, array $user_groups): a
...
+                WHERE t.forum_id = ?
+                  AND (t.topic_creator = ? OR t.sticky = 1)
+                ORDER BY p.created, p.post_date DESC LIMIT 1
+            SQL,
+            [$forumId, $userId]
+        );
+
+        if ($latest_post->count() && $latest_post = $latest_post->first()) {
+            return [
+                $latest_post->created ?? strtotime($latest_post->post_date),
```
Comment (tadhgboyle) [inline modules/Forum/classes/Forum.php:835]: Out of scope for this PR but this definitely feels like code smell

**Review by tadhgboyle — APPROVED** [review]

File: modules/Forum/classes/Forum.php (hunk 1 above)
Comment (Derkades) [reply modules/Forum/classes/Forum.php:835]: Can I mark this conversation as resolved to be able to merge this PR?

**Comment by Derkades** [comment]
please squash when merging, easier to backport :)

**Comment by samerton** [comment]
> please squash when merging, easier to backport :)

Absolutely - I'll always squash if there is >1 commit in the diff, if cherry-picking there is no difference either way

**Comment by Derkades** [comment]
Squashing is still useful even when a merge request only contains a single commit, because:
- the commit title is changed to the PR title, with a reference to the PR number (this means the backport commit will show up in the PR, as a confirmation that the backport has been done)
- the date of the commit is updated to be the date of the PR merge
- there is no additional merge commit which in my opinion is only clutter in this case

**Pull request merged**